backend/apps/matching/models/participant.py

```diff
@@ -48,6 +49,29 @@ def new(cls, participant_type, email, **kwargs):
     def p_type(self):
         return self.participant_type
 
+    @transaction.atomic
+    def change_approval(self, approver):
+        p_type = self.p_type().lower()
+
+        # check permission only here in case it was forgotten in the view
+        # maybe we can give a nicer message here?
+        if not approver.has_perm("matching.perm_approve_%s"):
```

Comment:
Building permissions like this seems error prone. Also there seems to be a `% p_type` missing?

Reply:
yes, thanks :)